display_information/update-graphics/src/main/java/com/esri/samples/update_graphics/UpdateGraphicsSample.java

```diff
@@ -167,39 +159,68 @@ public void start(Stage stage) {
-            // add new location to selected graphic
-            Point mapPoint = mapView.screenToLocation(mapViewPoint);
-            selectedGraphic.setGeometry(mapPoint);
-            isUpdateLocationActive = false;
-          } else {
-            // identify the graphic that was selected
-            identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
-
-            identifyGraphics.addDoneListener(() -> {
-              try {
-                if (!identifyGraphics.get().getGraphics().isEmpty()) {
-                  // store the selected graphic
-                  selectedGraphic = identifyGraphics.get().getGraphics().get(0);
-                  selectedGraphic.setSelected(true);
-                  String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
-                  symbolBox.getSelectionModel().select(style);
-
-                  enableUI(false);
-                } else {
-                  enableUI(true);
-                }
-              } catch (Exception x) {
-                new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+          mapViewPoint = new Point2D(e.getX(), e.getY());
+
+          // identify graphics on the graphics overlay
+          identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
+
+          identifyGraphics.addDoneListener(() -> {
+            try {
+              if (!identifyGraphics.get().getGraphics().isEmpty()) {
+                // get the first identified graphic
+                selectedGraphic = identifyGraphics.get().getGraphics().get(0);
+                // select the identified graphic
+                selectedGraphic.setSelected(true);
+                // update the drop down box with the identified graphic's current symbol
+                String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
+                symbolBox.getSelectionModel().select(style);
+                // show the UI
+                disableUI(false);
+              } else {
+                disableUI(true);
               }
-            });
-          }
+            } catch (Exception x) {
+              new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+            }
+          });
+        }
+      });
+
+      mapView.setOnMouseDragged(e -> {
+        if (selectedGraphic.isSelected()) {
+          // set the cursor to the move
+          mapView.setCursor(Cursor.MOVE);
+
+          // disable to UI while moving the graphic
+          disableUI(true);
+
+          // create a point from the dragged location
+          mapViewPoint = new Point2D(e.getX(), e.getY());
+          Point mapPoint = mapView.screenToLocation(mapViewPoint);
+
+          // update the location of the graphic to the dragged location
+          selectedGraphic.setGeometry(mapPoint);
+          mapView.requestFocus();
```

Comment:
I had to look this one up a bit. Not sure which element of the SV takes focus away from the map view, but I don't think we should include the code in the sample if it's just a workaround for the SV. Can't provide an alternative off the top of my head, but we might want to do a session to check what we can do to avoid adding it here.

Reply:
Based on this [link](https://docs.oracle.com/javafx/2/events/processing.htm), for key events the target is the node that has focus, but for mouse events the target is the node at the location of the cursor. I tried to use `stackpane.onKeyPressed()` like the Create and Save KML File sample, but I didn't have any luck.

Would you recommended scrapping the ESC key functionality? A second click will de-select and stop moving the point.

Reply:
That's probably a good idea, considering the ESC functionality it an added bonus, but that extra line of code is a bit confusing!